feat(DMS): add datasource rocketmq topics (#3667)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/dms_rocketmq_topics.md

---
subcategory: "Distributed Message Service (DMS)"
---

# huaweicloud_dms_rocketmq_topics

Use this data source to get the list of DMS rocketMQ topics.

## Example Usage

```hcl
var "instance_id" {}

data "huaweicloud_dms_rocketmq_topics" "test" {
  instance_id           = var.instance_id
  name                  = "topic1"
  total_read_queue_num  = 3
  total_write_queue_num = 3
  permission            = "all"
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `instance_id` - (Required, String) Specifies the ID of the rocketMQ instance.

* `name` - (Optional, String) Specifies the topic name.

* `total_read_queue_num` - (Optional, Int) Specifies the number of total read queue.

* `total_write_queue_num` - (Optional, Int) Specifies the number of total write queue.

* `permission` - (Optional, String) Specifies the permission. Value options: **sub**, **pub** or **all**.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The data source ID.

> * `topics` - Lists the topics.

Reviewer: ```suggestion
* `topics` - The list of topics.
```

Author: done

---
subcategory: "Distributed Message Service (DMS)"
---

# huaweicloud_dms_rocketmq_topics

> Use this data source to get the list of DMS rocketMQ topic.

Reviewer 2: ```suggestion
Use this data source to get the list of DMS rocketMQ topics.
```

Author: done

## Example Usage

```hcl
var "instance_id" {}

data "huaweicloud_dms_rocketmq_topics" "test" {
  instance_id = var.instance_id
  name                  = "topic1"
  total_read_queue_num  = 3
  total_write_queue_num = 3
  permission            = "all"

> }

Reviewer 2: align the equal signs

Author: done

```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `instance_id` - (Required, String) Specifies the ID of the rocketMQ instance.

* `name` - (Optional, String) Specifies the topic name.

* `total_read_queue_num` - (Optional, Int) Specifies the number of total read queue.

* `total_write_queue_num` - (Optional, Int) Specifies the number of total write queue.

> * `permission` - (Optional, String) Specifies the permission, the possible value is sub, pub or all.

Reviewer 2: ```suggestion
* `permission` - (Optional, String) Specifies the permission. Value options: **sub**, **pub** or **all**.
```

Author: done

* `message_type` (Optional, String) Specifies the message_type, which will be supported in RocketMQ instance V5.x.

>   The possible value is NORMAL, FIFO, DELAY or TRANSACTION

Reviewer 2: delete it

Author: done

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> 

Reviewer 2: the id is missing

Author: done

* `topics` - Indicates the list of the topic.

The `topics` block contains:

* `name` - Specifies the topic name.

> * `total_read_queue_num` - (Int) Specifies the number of total read queue.

Reviewer 2: ```suggestion
* `total_read_queue_num` - Indicates the number of total read queue.
```
fix other places

Author: done

* `total_write_queue_num` - (Int) Specifies the number of total write queue.

* `permission` - (String) Specifies the permission, the possible value is sub, pub or all.

* `message_type` (String) Specifies the message_type, which will be supported in RocketMQ instance V5.x.

>   The possible value is NORMAL, FIFO, DELAY or TRANSACTION.

Reviewer 2: delete it

Author: done

  instance_id = var.instance_id
  name                  = "topic1"
  total_read_queue_num  = 3
  total_write_queue_num = 3
  permission            = "all"
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `instance_id` - (Required, String) Specifies the ID of the rocketMQ instance.

* `name` - (Optional, String) Specifies the topic name.

* `total_read_queue_num` - (Optional, Int) Specifies the number of total read queue.

* `total_write_queue_num` - (Optional, Int) Specifies the number of total write queue.

* `permission` - (Optional, String) Specifies the permission, the possible value is sub, pub or all.

* `message_type` (Optional, String) Specifies the message_type, which will be supported in RocketMQ instance V5.x.
  The possible value is NORMAL, FIFO, DELAY or TRANSACTION

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `topics` - Lists the topics.
  The [topics](#dms_rockermq_topics) structure is documented below.

> <a name="dms_rockermq_topics"></a>

Reviewer 2: ```suggestion
  The [topics](#DMS_rockermq_topics) structure is documented below.

<a name="DMS_rockermq_topics"></a>
```

Author: done

The `topics` block supports:

* `name` - Specifies the topic name.

* `total_read_queue_num` - (Int) Specifies the number of total read queue.

* `total_write_queue_num` - (Int) Specifies the number of total write queue.

* `permission` - (String) Specifies the permission, the possible value is sub, pub or all.

* `message_type` (String) Specifies the message_type, which will be supported in RocketMQ instance V5.x.
  The possible value is NORMAL, FIFO, DELAY or TRANSACTION.

* `brokers` - Lists the brokers.
  The [brokers](#dms_rockermq_topic_brokers) structure is documented below.

<a name="dms_rockermq_topic_brokers"></a>

> The `brokers` block supports:

Reviewer 2: ```suggestion
  The [brokers](#DMS_rockermq_topic_brokers) structure is documented below.

<a name="DMS_rockermq_topic_brokers"></a>
The `brokers` block supports:
```

Author: done
